core/jvm/src/main/scala/zio/internal/FiberInbox.scala

```diff
@@ -0,0 +1,15 @@
+package zio.internal
+
+import org.jctools.queues.atomic.unpadded.MpscLinkedAtomicUnpaddedQueue
+
+private[zio] final class FiberInbox extends Serializable {
+  private val inbox = new MpscLinkedAtomicUnpaddedQueue[FiberMessage]()
+
+  def poll(): FiberMessage = inbox.relaxedPoll()
```

Comment:
I've seen you changed this to use `relaxedPoll`. We definitely don't want that. We need to be selective on where to use it, and _I think_ the only place that it really makes sense is in the `drainQueueWhileRunning` method. Also see my comment [here](https://github.com/zio/zio/pull/9036#discussion_r1701422436)